File: code_samples/kotlin/web_development_rest_api.kt
/**
 * Web Development: Rest Api
 * AI/ML Training Sample
 */

class RestApi {
    var data: String = ""
        private set
    
    fun process(input: String) {
        data = input
    }
    
    fun getData(): String = data
    
    fun validate(): Boolean = data.isNotEmpty()
}

fun main() {
    val instance = RestApi()
    instance.process("example")
    println("Data: ${instance.getData()}")
    println("Valid: ${instance.validate()}")
}

File: code_samples/dart/web_development_rest_api.dart
// Web Development: Rest Api
// AI/ML Training Sample

class RestApi {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = RestApi();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
